### PR TITLE
docs: improve the documentation framework.

### LINKED ISSUE #197: Add api docs.
We shall provide public api doc.

### PR DESCRIPTION
**Pull request by zhouaihui**
This will close #197.

- [x] Docs framework
- [ ] Why RayFed
- [ ] Quick start
- [ ] Tutorials
- [x] API
- [ ] Architecture of RayFed
- [ ] Principle of RayFed

### CODE REVIEW

File: docs/source/advanced_topic/architecture.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# The architecture of RayFed
```
Comment (jovany-wang) [inline docs/source/advanced_topic/architecture.md:1]: Should this be?
```suggestion
# The Architecture of RayFed
```

**Review by jovany-wang — APPROVED** [review]
LGTM

File: docs/source/advanced_topic/architecture.md (hunk 1 above)
Comment (zhouaihui) [reply docs/source/advanced_topic/architecture.md]: Nice catch

**Comment by fengsp** [comment]
Maybe we should rename `.md` files to `.rst`

**Comment by zhouaihui** [comment]
rst and md are all ok. 
For me, markdown +1.

**Pull request merged**